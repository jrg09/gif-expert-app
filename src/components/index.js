//archivo de barril, archivo Ã­ndice
export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GifItem";
export * from "../hooks/useFetchGifs";
