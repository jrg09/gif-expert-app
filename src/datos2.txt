este s n nuevo archivo.