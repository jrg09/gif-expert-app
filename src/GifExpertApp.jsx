import { useRef, useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Goku"]);

  const onNewCategory = (event) => {
    categories.filter((category) => category == event).length == 0 &&
      setCategories([event, ...categories]);
  };

  return (
    <>
      <h1>Gif Expert App</h1>

      <AddCategory onNewCategory={(event) => onNewCategory(event)} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
