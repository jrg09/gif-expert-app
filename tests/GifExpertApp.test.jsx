import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en <GifExpertApp />", () => {
  test("01 debe hacer match con el snapshot", () => {
    const { container } = render(<GifExpertApp />);
    //screen.debug();
    expect(container).toMatchSnapshot();
  });

  test("02 debe esperar y mostrar el titulo de Goku y las imagenes resultado", async () => {
    const { container } = render(<GifExpertApp />);
    await waitFor(() => {
      expect(container.getElementsByClassName("card").length).toBeGreaterThan(
        3
      );
    });

    //screen.debug();
    expect(screen.getByText("Goku")).toBeTruthy();
    expect(container.getElementsByClassName("card").length).toBeGreaterThan(3);
  });

  test("03 debe escribir algo en input y hacer submit al formulario", () => {
    render(<GifExpertApp />);

    screen.debug();

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    const category = "stonks";
    fireEvent.input(input, { target: { value: category } });

    fireEvent.submit(form);
    screen.debug();

    expect(input.value).toBe("");

    // expect(onNewCategory).toHaveBeenCalled();
    // expect(onNewCategory).toHaveBeenCalledTimes(1);
    // expect(onNewCategory).toHaveBeenCalledWith(category);
  });
});
