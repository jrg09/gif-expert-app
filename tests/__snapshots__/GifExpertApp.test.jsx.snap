// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> 01 debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Expert App
  </h1>
  <form
    aria-label="formulario"
  >
    <input
      placeholder="Buscar GIFs"
      type="text"
      value=""
    />
  </form>
  <div>
    <h4>
      Goku
    </h4>
    <div>
      <svg
        data-testid="infinity-spin"
        height="100"
        viewBox="0 0 200 100"
        width="200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="sc-bcXHqe fbAoTF"
          d="M93.9,46.4c9.3,9.5,13.8,17.9,23.5,17.9s17.5-7.8,17.5-17.5s-7.8-17.6-17.5-17.5c-9.7,0.1-13.3,7.2-22.1,17.1 c-8.9,8.8-15.7,17.9-25.4,17.9s-17.5-7.8-17.5-17.5s7.8-17.5,17.5-17.5S86.2,38.6,93.9,46.4z"
          data-testid="infinity-spin-path-1"
          fill="none"
          stroke="#4fa94d"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-miterlimit="10"
          stroke-width="4"
        />
        <path
          d="M93.9,46.4c9.3,9.5,13.8,17.9,23.5,17.9s17.5-7.8,17.5-17.5s-7.8-17.6-17.5-17.5c-9.7,0.1-13.3,7.2-22.1,17.1 c-8.9,8.8-15.7,17.9-25.4,17.9s-17.5-7.8-17.5-17.5s7.8-17.5,17.5-17.5S86.2,38.6,93.9,46.4z"
          data-testid="infinity-spin-path-2"
          fill="none"
          opacity="0.07"
          stroke="#4fa94d"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-miterlimit="10"
          stroke-width="4"
        />
      </svg>
    </div>
  </div>
</div>
`;
